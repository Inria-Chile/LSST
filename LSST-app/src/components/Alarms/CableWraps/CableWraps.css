.cable-wraps-container {
    display: flex;
    flex-direction: column;
    margin: 5px;
    height: 100%;  
    margin-top: 0;
}

.cable-wraps-container .cable-wraps-content {
    color: #96a3b3;
    background: var(--theme-div-background);
    margin: 5px;
    height: 100%;
    margin-top: 0;
    padding-right: 5px;
    padding-left: 20px;
}

.rotator-diff {
    font-size: 17px;
    margin-bottom: 0;
    margin-left: 10px;
}

.rotator-diff-value {
    font-weight: bold;
    margin-left: 10px;
}

.cam-cable {
    max-width: 20%;
}

.cable-wrap-separator {
    height: 290px;
    width: 3px;
    margin-top: 40px;
    margin-right: 50px;
    border-radius: 10px;
    background: #2d3d4e;
}