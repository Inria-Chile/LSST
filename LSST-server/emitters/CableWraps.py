import sys
import time
import datetime
import random

from flask_socketio import send, emit
from flask_socketio import SocketIO

def start_listening_fake_cable_wraps(app, socketio):

    print('Emitting fake cable wraps')
    initial_date = 10000
    date_step = 1000
    az_cable_position = 0
    az_rotator_position = 0
    cam_cable_position = 0
    cam_rotator_position = 0
    socketio.emit('cable_wraps', {'cam_cable': cam_cable_position, 'cam_rot': cam_rotator_position, 'az_cable': az_cable_position, 'az_rot': az_rotator_position})
         
    while True:
        time.sleep(1)
        with app.test_request_context('/'):
            
            az_cable_wrap = random.uniform(-5,5)
            az_cable_position = az_cable_position+az_cable_wrap
            az_rot_wrap = random.uniform(-2,2)+az_cable_wrap
            az_rotator_position = az_rotator_position+az_rot_wrap

            cam_cable_wrap = random.uniform(-5,5)
            cam_cable_position = cam_cable_position+cam_cable_wrap
            cam_rot_wrap = random.uniform(-2,2)+cam_cable_wrap
            cam_rotator_position = cam_rotator_position+cam_rot_wrap
            
            socketio.emit('cable_wraps', {'cam_cable': cam_cable_position, 'cam_rot': cam_rotator_position, 'az_cable': az_cable_position, 'az_rot': az_rotator_position})
            initial_date = initial_date + date_step
