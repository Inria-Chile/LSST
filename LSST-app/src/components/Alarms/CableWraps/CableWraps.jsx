import React, { Component } from 'react'
import './CableWraps.css';
// import DraggableTitle from '../Utils/DraggableTitle';
import AZCableWrap from './AZCableWrap/AZCableWrap';
import CameraCableWrap from './CameraCableWrap/CameraCableWrap';
import DraggableTitle from '../../Utils/DraggableTitle';
import * as d3 from 'd3';
import openSocket from 'socket.io-client';

class CableWraps extends Component {

    constructor(props) {
        super(props);
        this.socket = openSocket(window.location.origin + '');
        this.state = {
            cable_wraps: null
        };

    }

    componentDidMount() {
        this.socket.on('cable_wraps', timestamp => this.receiveMsg(timestamp));

    }

    receiveMsg(msg) {
        console.log(msg);
        this.setState({
            cable_wraps: msg
        })

    }

    drawBackground(g, radio, tau, arc) {
        g.append("circle")
            .attr("cx", 0)
            .attr("cy", 0)
            .attr("r", radio - 5)
            .style("fill", "#384b5f");

        g.append("circle")
            .attr("cx", 0)
            .attr("cy", 0)
            .attr("r", radio - 70)
            .style("fill", "#7296b0");

        g.append("path")
            .datum({ endAngle: tau })
            .style("fill", "#4d667b")
            .attr("d", arc);
    }

    drawLimits(g, radio, start, end) {
        g.append("rect")
            .attr("x", 0)
            .attr("y", -radio - 10)
            .attr("width", 5)
            .attr("height", 30)
            .style("fill", "#ffffff")

        g.append("rect")
            .attr("x", -radio - 10)
            .attr("y", 0)
            .attr("width", 30)
            .attr("height", 2)
            .style("fill", "#ffffff")

        g.append("text")
            .attr("x", -radio - 50)
            .attr("y", 5)
            .text(start + "°")
            .style("fill", "#ffffff")

        g.append("rect")
            .attr("x", radio - 20)
            .attr("y", 0)
            .attr("width", 30)
            .attr("height", 2)
            .style("fill", "#ffffff")

        g.append("text")
            .attr("x", radio + 15)
            .attr("y", 5)
            .text(end + "°")
            .style("fill", "#ffffff")
    }

    arcTween(newAngle, arc) {
        return function (d) {
            var interpolate = d3.interpolate(d.endAngle, newAngle);
            return function (t) {
                d.endAngle = interpolate(t);
                return arc(d);
            };
        };
    }


    render() {
        return (
            <div className="cable-wraps-container">
                <DraggableTitle title='Cable Wraps' />
                <div className="cable-wraps-content pull-left">
                    <div className="row">
                        <div className="cam-cable col-md-6">
                            <h4>Camera Cable Wrap</h4>
                            {
                                this.state.cable_wraps ?
                                    <p className='rotator-diff'>Rotator angle difference:
                                    <span className='rotator-diff-value'>
                                            {(this.state.cable_wraps.camera.cable - this.state.cable_wraps.camera.rotator).toFixed(2) + 'º'}
                                        </span>
                                    </p> :
                                    <p></p>
                            }
                            <CameraCableWrap
                                height={300}
                                width={400}
                                drawBackground={this.drawBackground}
                                drawLimits={this.drawLimits}
                                arcTween={this.arcTween}
                                cable_wrap={(this.state.cable_wraps) ? this.state.cable_wraps.camera : null} />
                        </div>
                        <div className="az-cable col-md-6">
                            <h4>Azimuth Cable Wrap</h4>
                            {
                                this.state.cable_wraps ?
                                    <p className='rotator-diff'>Rotator angle difference:
                                    <span className='rotator-diff-value'>
                                            {(this.state.cable_wraps.az.cable - this.state.cable_wraps.az.rotator).toFixed(2) + 'º'}
                                        </span>
                                    </p> :
                                    <p></p>
                            }
                            <AZCableWrap
                                height={300}
                                width={400}
                                drawBackground={this.drawBackground}
                                drawLimits={this.drawLimits}
                                arcTween={this.arcTween}
                                cable_wrap={(this.state.cable_wraps) ? this.state.cable_wraps.az : null} />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default CableWraps;
