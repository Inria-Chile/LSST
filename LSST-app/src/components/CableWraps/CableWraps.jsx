import React, { Component } from 'react'
import './CableWraps.css';
import DraggableTitle from '../Utils/DraggableTitle';
import AZCableWrap from './AZCableWrap/AZCableWrap';
import CameraCableWrap from './CameraCableWrap/CameraCableWrap';
import * as d3 from 'd3';
import openSocket from 'socket.io-client';

class CableWraps extends Component {

    constructor(props){
        super(props);
        this.socket = openSocket(window.location.origin+'');
        this.state({
            cable_wraps: null
        })
        
    }

    drawBackground(g,radio, tau, startAngle){

        let arc = d3.arc()
        .innerRadius(radio-10)
        .outerRadius(radio)
        .startAngle(startAngle);
        
        g.append("circle")
            .attr("cx", 0)
            .attr("cy", 0)
            .attr("r", radio-5)
            .style("fill", "#384b5f");

        g.append("circle")
            .attr("cx", 0)
            .attr("cy", 0)
            .attr("r", radio-70)
            .style("fill", "#7296b0");

        g.append("path")
            .datum({endAngle: tau})
            .style("fill", "#4d667b")
            .attr("d", arc);

        g.append("rect")
            .attr("x", 0)
            .attr("y", -radio-10)
            .attr("width", 5)
            .attr("height", 30)
            .style("fill", "#ffffff")
        
       
    }

    componentDidMount(){
        this.socket.on('cable_wraps', timestamp => this.receiveMsg(timestamp));
        
    }

    receiveMsg(msg){
        console.log("msg recieved!")
        console.log(msg)
        this.setState({
            cable_wraps:msg
        })

    }


    render() {
        return(
            <div className="cable-wraps-container">
            <DraggableTitle title='Cable Wraps'/>
            <div className="container pull-left">
            <div className="row">
                <div className="cam-cable col-md-6">
                <h4>CameraCable Wrap</h4>
                <CameraCableWrap height={300} width={300} drawBackground={this.drawBackground}/>
                </div>
                <div className="az-cable col-md-6">
                <h4>Azimuth Cable Wrap</h4>
                <AZCableWrap height={300} width={300} drawBackground={this.drawBackground}/>
                </div>
            
            </div></div></div>
        );
    }
}
export default CableWraps;
