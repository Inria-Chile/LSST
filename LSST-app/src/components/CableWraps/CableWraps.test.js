import React from 'react';
import CableWraps from './CableWraps';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';


/*---- Snapshot testing ----*/
describe('CableWraps_Test',function(){

    it('renders correctly with shallow',()=>{
        const renderer = new shallowRenderer();
        const tree = renderer.render(<CableWraps/>);
        expect(tree).toMatchSnapshot();
    })


    /*it('renders correctly without shallow', ()=>{
        const tree = renderer
            .create(<CableWraps/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })*/
})



/*---- Other test ----*/
