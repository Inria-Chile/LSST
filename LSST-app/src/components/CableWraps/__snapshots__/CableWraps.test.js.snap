// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CableWraps_Test renders correctly with shallow 1`] = `
<div
  className="cable-wraps-container"
>
  <DraggableTitle
    title="Cable Wraps"
  />
  <div
    className="container pull-left"
  >
    <div
      className="row"
    >
      <div
        className="cam-cable col-md-6"
      >
        <h4>
          Camera Cable Wrap
        </h4>
        <CameraCableWrap
          arcTween={[Function]}
          cable_wrap={null}
          drawBackground={[Function]}
          drawLimits={[Function]}
          height={300}
          width={400}
        />
      </div>
      <div
        className="az-cable col-md-6"
      >
        <h4>
          Azimuth Cable Wrap
        </h4>
        <AZCableWrap
          arcTween={[Function]}
          cable_wrap={null}
          drawBackground={[Function]}
          drawLimits={[Function]}
          height={300}
          width={400}
        />
      </div>
    </div>
  </div>
</div>
`;
